src/ORM/Search/SearchContext.php

```diff
@@ -104,7 +105,18 @@ public function __construct($modelClass, $fields = null, $filters = null)
      */
     public function getSearchFields()
     {
-        return ($this->fields) ? $this->fields : singleton($this->modelClass)->scaffoldSearchFields();
```

Comment:
This was previously pointless, since `$this->fields` was _always_ being set to an empty `FieldList()` in the constructor. We therefore never got to the `scaffoldSearchFields()` call.